dask/array/tests/test_routines.py

```diff
@@ -502,7 +502,7 @@ def test_bincount_with_weights():
 
     dweights = da.from_array(weights, chunks=2)
     e = da.bincount(d, weights=dweights, minlength=6)
-    assert_eq(e, np.bincount(x, weights=dweights, minlength=6))
+    assert_eq(e, np.bincount(x, weights=dweights.compute(), minlength=6))
```

Comment:
Why was this change needed?

Reply:
Because in this, the right argument of `assert_eq()` is used as baseline, and `x` being a NumPy array, it fails here:

```python
    @wraps(np.bincount)
    def bincount(x, weights=None, minlength=None):
        if minlength is None:
            raise TypeError("Must specify minlength argument in da.bincount")
        assert x.ndim == 1
        if weights is not None:
>           assert weights.chunks == x.chunks
E           AttributeError: 'numpy.ndarray' object has no attribute 'chunks'
```

Therefore, to me it makes sense to make both arguments NumPy arrays and let it fallback to NumPy's implementation.

Reply:
I think the reverse should be done, with `weights` being coerced to `da.Array` with the same chunks as `x`.

Reply:
Ah I see.  I think I agree with @pentschev here.  The left argument, `e`, is testing the dask/dask situation while the right argument is testing the numpy/numpy situation.

Reply:
> I think the reverse should be done, with `weights` being coerced to `da.Array` with the same chunks as `x`.

That would cause `da.bincount()` to be dispatched, then we would be testing against `e` which computes exactly that.